In [151]:
import pandas as pd
import numpy as np

In [152]:
df_centro_oeste = pd.read_csv("historico_regiao_centro-oeste.csv")
df_nordeste = pd.read_csv("historico_regiao_nordeste.csv")
df_norte = pd.read_csv("historico_regiao_norte.csv")
df_sudeste = pd.read_csv("historico_regiao_sudeste.csv")
df_sul = pd.read_csv("historico_regiao_sul.csv")

In [153]:
df_centro_oeste.head()

Unnamed: 0 Janeiro Fevereiro Março Abril  Maio  Junho  Julho  Agosto  \
0       1998       -         -     -     -     -   2456   4600   17896   
1       1999     292       433   119   198  1034   2581   5538   24132   
2       2000     380       127    87   165  1379   4738   1842    7523   
3       2001     169       154    73   466  1388   7103   2536   11018   
4       2002     489       129   266   950  2782   8709   8826   22641   

   Setembro  Outubro Novembro Dezembro  Total  
0     13718     4320      537      190  43717  
1     16607     8918     2149      373  62374  
2      8057     7643      802      243  32986  
3     11745     7584      608      199  43043  
4     26090    15136     5459     1732  93209

In [154]:
# Remomea coluna referente aos anos
df_centro_oeste.rename(columns={'Unnamed: 0':'Ano'}, inplace=True)
df_nordeste.rename(columns={'Unnamed: 0':'Ano'}, inplace=True)
df_norte.rename(columns={'Unnamed: 0':'Ano'}, inplace=True)
df_sudeste.rename(columns={'Unnamed: 0':'Ano'}, inplace=True)
df_sul.rename(columns={'Unnamed: 0':'Ano'}, inplace=True)

In [155]:
df_centro_oeste.shape
#df_nordeste.shape

(27, 14)

In [156]:
# Remove linhas referentes aos anos de 1998, 1999 e 2021 e as linhas 'máximo', 'média' e 'mínimo'
df_centro_oeste.drop([0,1,26,25,24,23], axis=0, inplace=True)
df_nordeste.drop([0,1,26,25,24,23], axis=0, inplace=True)
df_norte.drop([0,1,26,25,24,23], axis=0, inplace=True)
df_sudeste.drop([0,1,26,25,24,23], axis=0, inplace=True)
df_sul.drop([0,1,26,25,24,23], axis=0, inplace=True)

# Reinicia os índices no próprio dataframe e não adiciona os índices antigos em uma nova coluna
df_centro_oeste.reset_index(inplace=True, drop=True)
df_nordeste.reset_index(inplace=True, drop=True)
df_norte.reset_index(inplace=True, drop=True)
df_sudeste.reset_index(inplace=True, drop=True)
df_sul.reset_index(inplace=True, drop=True)

In [157]:
# Substitui todos os caracteres '-' no dataframe por 0
#df_centro_oeste.replace('-', 0, inplace=True)
#df_nordeste.replace('-', 0, inplace=True)
#df_norte.replace('-', 0, inplace=True)
#df_sudeste.replace('-', 0, inplace=True)
#df_sul.replace('-', 0, inplace=True)

In [158]:
# Verificação dos tipos de dados
df_centro_oeste.dtypes

Ano          object
Janeiro      object
Fevereiro    object
Março        object
Abril        object
Maio         object
Junho         int64
Julho         int64
Agosto        int64
Setembro      int64
Outubro       int64
Novembro     object
Dezembro     object
Total         int64
dtype: object

In [159]:
df_nordeste.dtypes

Ano          object
Janeiro      object
Fevereiro    object
Março        object
Abril        object
Maio         object
Junho         int64
Julho         int64
Agosto        int64
Setembro      int64
Outubro       int64
Novembro     object
Dezembro     object
Total         int64
dtype: object

In [160]:
# Ajuste dos tipos de dados

df_centro_oeste['Ano'] = df_centro_oeste['Ano'].astype('int16')
df_nordeste['Ano'] = df_nordeste['Ano'].astype('int16')
df_norte['Ano'] = df_norte['Ano'].astype('int16')
df_sudeste['Ano'] = df_sudeste['Ano'].astype('int16')
df_sul['Ano'] = df_sul['Ano'].astype('int16')

# Demais colunas...
colunas = df_centro_oeste.columns[1:]

for coluna in colunas:
    df_centro_oeste[coluna] = df_centro_oeste[coluna].astype('int32')
    df_nordeste[coluna] = df_nordeste[coluna].astype('int32')
    df_norte[coluna] = df_norte[coluna].astype('int32')
    df_sudeste[coluna] = df_sudeste[coluna].astype('int32')
    df_sul[coluna] = df_sul[coluna].astype('int32')

In [161]:
# Verificação dos tipos após modificação
df_centro_oeste.dtypes

Ano          int16
Janeiro      int32
Fevereiro    int32
Março        int32
Abril        int32
Maio         int32
Junho        int32
Julho        int32
Agosto       int32
Setembro     int32
Outubro      int32
Novembro     int32
Dezembro     int32
Total        int32
dtype: object

In [162]:
df_nordeste.dtypes

Ano          int16
Janeiro      int32
Fevereiro    int32
Março        int32
Abril        int32
Maio         int32
Junho        int32
Julho        int32
Agosto       int32
Setembro     int32
Outubro      int32
Novembro     int32
Dezembro     int32
Total        int32
dtype: object

In [163]:
df_centro_oeste.set_index('Ano', inplace=True)
df_nordeste.set_index('Ano', inplace=True)
df_norte.set_index('Ano', inplace=True)
df_sudeste.set_index('Ano', inplace=True)
df_sul.set_index('Ano', inplace=True)

In [164]:
df_centro_oeste.head()

Janeiro  Fevereiro  Março  Abril  Maio  Junho  Julho  Agosto  Setembro  \
Ano                                                                            
2000      380        127     87    165  1379   4738   1842    7523      8057   
2001      169        154     73    466  1388   7103   2536   11018     11745   
2002      489        129    266    950  2782   8709   8826   22641     26090   
2003     1134        764    734   1234  3304  10424  13799   14861     23121   
2004      651        682   1204   1207  4435  12600  14710   19358     40818   

      Outubro  Novembro  Dezembro   Total  
Ano                                        
2000     7643       802       243   32986  
2001     7584       608       199   43043  
2002    15136      5459      1732   93209  
2003     8700      3131      2001   83207  
2004    14079      3493      1715  114952

In [165]:
df_nordeste.head()

Janeiro  Fevereiro  Março  Abril  Maio  Junho  Julho  Agosto  Setembro  \
Ano                                                                            
2000       72        106    162     99   142    318    531    2036      5055   
2001      121        116      7     43   292    342   1317    4485     12226   
2002      270        630    306    253   350    523   2901    8169     16617   
2003     2326        516    571    739   563   1470   4056    4667     22278   
2004      963        306    377    371   737   1402   2661    5113     17932   

      Outubro  Novembro  Dezembro  Total  
Ano                                       
2000    10018      3640      2023  24202  
2001    14927      8736      2413  45025  
2002    18400     15334      8249  72002  
2003    20040     16441      9669  83336  
2004    16286     19339     13306  78793

In [166]:
# Definindo novo dataframe 'df_comparativo_anual'
colunas_comparativo = ['Região']
i = 0

for ano in range(2000,2021,1):
    colunas_comparativo.append(ano)
    if ano < 2020:
        i = ano%1000
        colunas_comparativo.append('%' + '('+ str(i) + '-' + str(i+1) + ')')

df_comparativo_anual = pd.DataFrame(np.zeros((5,42),np.int32),columns=colunas_comparativo)

In [167]:
# Definição dos índices (linhas)
regioes = 'Centro-Oeste Nordeste Norte Sudeste Sul'.split()

df_comparativo_anual['Região'] = regioes
df_comparativo_anual.set_index('Região', inplace=True)

In [168]:
df_comparativo_anual.head()

2000  %(0-1)  2001  %(1-2)  2002  %(2-3)  2003  %(3-4)  2004  \
Região                                                                       
Centro-Oeste     0       0     0       0     0       0     0       0     0   
Nordeste         0       0     0       0     0       0     0       0     0   
Norte            0       0     0       0     0       0     0       0     0   
Sudeste          0       0     0       0     0       0     0       0     0   
Sul              0       0     0       0     0       0     0       0     0   

              %(4-5)  ...  %(15-16)  2016  %(16-17)  2017  %(17-18)  2018  \
Região                ...                                                   
Centro-Oeste       0  ...         0     0         0     0         0     0   
Nordeste           0  ...         0     0         0     0         0     0   
Norte              0  ...         0     0         0     0         0     0   
Sudeste            0  ...         0     0         0     0         0     0   
Sul                0  ...         0     0         0     0         0     0   

              %(18-19)  2019  %(19-20)  2020  
Região                                        
Centro-Oeste         0     0         0     0  
Nordeste             0     0         0     0  
Norte                0     0         0     0  
Sudeste              0     0         0     0  
Sul                  0     0         0     0  

[5 rows x 41 columns]

In [169]:
# Atualização dos valores ano a ano para cada região
for ano in range(2000,2021,1):
    df_comparativo_anual.loc['Centro-Oeste', ano] = df_centro_oeste.loc[ano, 'Total']
    df_comparativo_anual.loc['Nordeste', ano] = df_nordeste.loc[ano, 'Total']
    df_comparativo_anual.loc['Norte', ano] = df_norte.loc[ano, 'Total']
    df_comparativo_anual.loc['Sudeste', ano] = df_sudeste.loc[ano, 'Total']
    df_comparativo_anual.loc['Sul', ano] = df_sul.loc[ano, 'Total']

In [170]:
df_comparativo_anual.columns

Index([      2000,   '%(0-1)',       2001,   '%(1-2)',       2002,   '%(2-3)',
             2003,   '%(3-4)',       2004,   '%(4-5)',       2005,   '%(5-6)',
             2006,   '%(6-7)',       2007,   '%(7-8)',       2008,   '%(8-9)',
             2009,  '%(9-10)',       2010, '%(10-11)',       2011, '%(11-12)',
             2012, '%(12-13)',       2013, '%(13-14)',       2014, '%(14-15)',
             2015, '%(15-16)',       2016, '%(16-17)',       2017, '%(17-18)',
             2018, '%(18-19)',       2019, '%(19-20)',       2020],
      dtype='object')

In [171]:
len(df_comparativo_anual.columns)

41

In [172]:
# Cálculo da diferença percentual
ano = 2000

for indice in range(1,40,2):
    coluna = df_comparativo_anual.columns[indice]
    df_comparativo_anual[coluna] = round(((df_comparativo_anual[ano + 1] - df_comparativo_anual[ano])/(df_comparativo_anual[ano]))*100, 1)
    ano += 1

In [173]:
df_comparativo_anual.head()

2000  %(0-1)   2001  %(1-2)    2002  %(2-3)    2003  %(3-4)  \
Região                                                                       
Centro-Oeste  32986    30.5  43043   116.5   93209   -10.7   83207    38.2   
Nordeste      24202    86.0  45025    59.9   72002    15.7   83336    -5.5   
Norte         32162    51.2  48614   154.5  123740    -0.6  122938    23.0   
Sudeste        8948   -20.9   7082   156.4   18157    49.3   27110   -31.5   
Sul            3232   -46.8   1720   731.7   14306    72.3   24646   -31.5   

                2004  %(4-5)  ...  %(15-16)   2016  %(16-17)   2017  %(17-18)  \
Região                        ...                                               
Centro-Oeste  114952   -20.6  ...       0.5  38819      11.7  43363     -45.6   
Nordeste       78793    -2.4  ...     -33.1  43201       3.4  44652     -24.8   
Norte         151230     8.7  ...     -16.1  81014      21.3  98247     -36.9   
Sudeste        18579    -0.9  ...     -14.4  10944      22.7  13431     -39.7   
Sul            16891   -32.1  ...     139.7  10239     -23.7   7815     -29.1   

               2018  %(18-19)   2019  %(19-20)   2020  
Região                                                 
Centro-Oeste  23610     112.6  50195      31.5  65992  
Nordeste      33591      31.5  44157      -7.5  40852  
Norte         62029      29.9  80559      12.3  90443  
Sudeste        8097      77.9  14407      10.7  15954  
Sul            5543      50.0   8314      14.9   9556  

[5 rows x 41 columns]

In [174]:
# Dataframe contendo apenas o total de focos por ano para cada região
df_total_focos = df_comparativo_anual.iloc[:, 0::2]

In [175]:
df_total_focos.head()

2000   2001    2002    2003    2004    2005    2006    2007  \
Região                                                                       
Centro-Oeste  32986  43043   93209   83207  114952   91276   53250  104231   
Nordeste      24202  45025   72002   83336   78793   76937   51555  104109   
Norte         32162  48614  123740  122938  151230  164458  113590  145950   
Sudeste        8948   7082   18157   27110   18579   18420   15802   29116   
Sul            3232   1720   14306   24646   16891   11472   14982   10509   

               2008   2009  ...   2011   2012   2013   2014   2015   2016  \
Região                      ...                                             
Centro-Oeste  39719  24391  ...  27138  41581  25324  33307  38620  38819   
Nordeste      60996  44118  ...  52038  72833  35628  44961  64539  43201   
Norte         87265  71099  ...  54370  82305  53010  74707  96568  81014   
Sudeste       15686   7360  ...  17414  13705   8442  17400  12780  10944   
Sul            8267   8134  ...   7139   6810   5741   5517   4271  10239   

               2017   2018   2019   2020  
Região                                    
Centro-Oeste  43363  23610  50195  65992  
Nordeste      44652  33591  44157  40852  
Norte         98247  62029  80559  90443  
Sudeste       13431   8097  14407  15954  
Sul            7815   5543   8314   9556  

[5 rows x 21 columns]

In [176]:
# Ano com maior concentração de queimadas para cada região 
for coluna in df_total_focos.T.columns:
    print(df_total_focos.T[df_total_focos.T[coluna] == df_total_focos.T[coluna].max()][coluna])

2004    114952
Name: Centro-Oeste, dtype: int32
2007    104109
Name: Nordeste, dtype: int32
2005    164458
Name: Norte, dtype: int32
2007    29116
Name: Sudeste, dtype: int32
2003    24646
Name: Sul, dtype: int32


In [177]:
# Ano com menor concentração de queimadas para cada região 
for coluna in df_total_focos.T.columns:
    print(df_total_focos.T[df_total_focos.T[coluna] == df_total_focos.T[coluna].min()][coluna])

2018    23610
Name: Centro-Oeste, dtype: int32
2000    24202
Name: Nordeste, dtype: int32
2000    32162
Name: Norte, dtype: int32
2001    7082
Name: Sudeste, dtype: int32
2001    1720
Name: Sul, dtype: int32


**Resumo dos anos com maior e menor número de focos de incêndio por região**

Região       | Maior número de focos    | Menor número de focos  |
:-------     | :---                     | :---                   |
Centro-Oeste | **2004** - 114952 focos  | **2018** - 23610 focos |
Nordeste     | **2007** - 104109 focos  | **2000** - 24202 focos |
Norte        | **2005** - 164458 focos  | **2000** - 32162 focos |
Sudeste      | **2007** - 29116  focos  | **2001** - 7082  focos |
Sul          | **2003** - 24646  focos  | **2001** - 1720  focos |

In [180]:
# Dataframe com as estatísticas descritivas para o total de focos dentro do período analisado para cada região

df_estatisticas_focos= pd.DataFrame(df_total_focos.mean(axis=1).apply(lambda x: int(x)), columns=['Média'])

df_estatisticas_focos['DP'] = df_total_focos.std(axis=1).apply(lambda x: int(x))
df_estatisticas_focos['Mediana'] = df_total_focos.median(axis=1).apply(lambda x: int(x))
df_estatisticas_focos['Máximo'] = df_total_focos.max(axis=1).apply(lambda x: int(x))
df_estatisticas_focos['Mínimo'] = df_total_focos.min(axis=1).apply(lambda x: int(x))
df_estatisticas_focos['Total'] = df_total_focos.sum(axis=1).apply(lambda x: int(x))

In [181]:
# Adiciona variação percentual por região em relação ao total de focos
df_estatisticas_focos['Total %'] = round(100 * (df_estatisticas_focos['Total']/df_estatisticas_focos['Total'].sum(axis=0)), 2)

In [182]:
df_estatisticas_focos.head()

Média     DP  Mediana  Máximo  Mínimo    Total  Total %
Região                                                               
Centro-Oeste  54695  28404    43043  114952   23610  1148614    23.81
Nordeste      57093  20357    51555  104109   24202  1198958    24.86
Norte         93394  35879    87265  164458   32162  1961282    40.66
Sudeste       15301   6009    15686   29116    7082   321323     6.66
Sul            9199   5176     8134   24646    1720   193180     4.01

**Análise:** 
* Dentro do período avaliado (anos 2000 a 2020), a região que concentrou a maior parte dos focos de incêndio foi a região Norte, respondendo por aproximadamente 41% do total do número de focos de incêndios no Brasil; 
* As regiões com menores concentrações de focos de incêndio são as regiões Sul (4.0%) e Sudeste (6.7%), respectivamente;
* Analisando-se os desvios-padrões, observa-se que os valores alcançados são de no mínimo 35% do valor médio (região Centro-Oeste) e no máximo 56% do valor médio (região Sul). Tais resultados demonstram ampla variabilidade anual das queimadas por região. Destaca-se a região Sul, na qual o desvio-padrão é maior que a metade da média (ou a metade da mediana) mesmo sendo esta a região com menor número de focos para o período avaliado;
* As regiões Centro-Oeste e Nordeste apresentaram resultados totais muito próximos para a quantidade de focos. Cada uma dessas regiões respondeu por aproximadamente 24% do total de queimadas para todo o território brasileiro. Excluindo-se a região Norte, essas duas regiões concentram a maior quantidade de focos e totalizam, aproximadamente, 50% dos focos de incêndios no Brasil;

In [183]:
# Dataframe contendo apenas as variações percentuais entre os anos para cada região
df_variacao_percentual = df_comparativo_anual.iloc[:, 1::2].T

In [184]:
df_variacao_percentual

Região    Centro-Oeste  Nordeste  Norte  Sudeste    Sul
%(0-1)            30.5      86.0   51.2    -20.9  -46.8
%(1-2)           116.5      59.9  154.5    156.4  731.7
%(2-3)           -10.7      15.7   -0.6     49.3   72.3
%(3-4)            38.2      -5.5   23.0    -31.5  -31.5
%(4-5)           -20.6      -2.4    8.7     -0.9  -32.1
%(5-6)           -41.7     -33.0  -30.9    -14.2   30.6
%(6-7)            95.7     101.9   28.5     84.3  -29.9
%(7-8)           -61.9     -41.4  -40.2    -46.1  -21.3
%(8-9)           -38.6     -27.7  -18.5    -53.1   -1.6
%(9-10)          229.6      84.6   78.6    205.6   -0.7
%(10-11)         -66.2     -36.1  -57.2    -22.6  -11.6
%(11-12)          53.2      40.0   51.4    -21.3   -4.6
%(12-13)         -39.1     -51.1  -35.6    -38.4  -15.7
%(13-14)          31.5      26.2   40.9    106.1   -3.9
%(14-15)          16.0      43.5   29.3    -26.6  -22.6
%(15-16)           0.5     -33.1  -16.1    -14.4  139.7
%(16-17)          11.7       3.4   21.3     22.7  -23.7
%(17-18)         -45.6     -24.8  -36.9    -39.7  -29.1
%(18-19)         112.6      31.5   29.9     77.9   50.0
%(19-20)          31.5      -7.5   12.3     10.7   14.9

In [186]:
df_variacao_percentual.describe()

Região  Centro-Oeste    Nordeste       Norte     Sudeste         Sul
count      20.000000   20.000000   20.000000   20.000000   20.000000
mean       22.155000   11.505000   14.680000   19.165000   38.205000
std        73.088292   46.222602   48.948024   71.959457  168.974388
min       -66.200000  -51.100000  -57.200000  -53.100000  -46.800000
25%       -38.725000  -29.025000  -21.600000  -27.825000  -25.050000
50%        13.850000    0.500000   16.800000  -14.300000   -8.100000
75%        41.950000   40.875000   32.650000   56.450000   18.825000
max       229.600000  101.900000  154.500000  205.600000  731.700000

**Análise**
* Entre os anos 2001 e 2002 a região sul apresentou uma elevação abrupta no número de focos de incêndios, aproximadamente 732%. Em 2001 havia sido contabilizado 1720 focos, ao passo que no ano de 2002 foram contabilizados 14306 focos. Nenhum outra região apresentou tamanho crescimento para um mesmo intervalo de tempo;
* Entre os anos de 2009 e 2010, as regiões Centro-Oeste e Sudeste apresentaram crescimento expressivo no número de focos de incêndios que superou os 200% em ambas as regiões;
* A menor redução no número de focos de incêndios em dois anos consecutivos ocorreu na região Centro-Oeste (-66,2%) entre os anos 2010 e 2011;
* No período de 2007 a 2009 observa-se redução no número de focos para todas as regiões. Em nenhum outro momento do período avaliado vericou-se tal ocorrência em intervalo de amplitude semelhante.